### ISSUE TITLE
Automaticlly set level in the sequence header (and/or allow specifying it)

### ISSUE THREAD

**Issue by dwbuiten** [xiph/rav1e#1313]
It's relatively easy to add a CLI/API option to specify it, and I can do that if you guys want.

However, it would be useful to set the level automatically. The only part I'm not sure about implementation-wise is the `MainMbps` and `HighMbps` entries in the levels tables. Do they make it impossible to derive the level before encoding is done?

**Comment by tdaede** [comment]
Yeah, they sorta do, which is why we don't set it yet. In practice we don't want to go back and rewrite the level, so we would actually want to either start dropping frames if we exceed the Mbps, or reencode them.

**Comment by dwbuiten** [comment]
Well that kinda sucks... how would you even manage to write a level when using e.g. a 'quality' mode like x264's CRF?

**Comment by tdaede** [comment]
Just pick one based on resolution and use the rate controller + frame dropping to keep it below the max in quality mode.

Alternately, do it the x264 crf way, which is to just write invalid bitstreams if you specify a crf too low... that's not ideal, obviously.

**Comment by dwbuiten** [comment]
Neither is particularily ideal, so maybe an option on which of the two bad choices to use...
